src/main/java/com/voicepin/flow/client/ExceptionMapper.java

```diff
@@ -1,59 +1,59 @@
 package com.voicepin.flow.client;
 
-
+import com.voicepin.flow.client.exception.AudioTooShortException;
 import com.voicepin.flow.client.exception.FlowServerException;
+import com.voicepin.flow.client.exception.InvalidAudioException;
+import com.voicepin.flow.client.exception.VoiceprintNotEnrolled;
 
-import javax.ws.rs.core.Response;
 import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.Collection;
 
-import static javax.ws.rs.core.Response.Status;
+import javax.ws.rs.core.Response;
 
 /**
  * @author mckulpa, kodrzywolek
  */
-public class ExceptionMapper {
-    private Map<Integer, Definition> map = new HashMap<>();
-
-    public ExceptionMapper() {
-        for (Definition definition : getDefinitions()) {
-            for (Integer code : definition.getCodes()) {
-                Definition oldValue = map.put(code, definition);
-                if (oldValue != null) {
-                    throw new RuntimeException("Tried to map two exceptions to the same error code (" + code + ")");
-                }
+class ExceptionMapper {
+
+    private final Collection<Definition> definitions = new ArrayList<>();
```

Comment:
" We follow two rules in the matter of optimization: Rule 1. Don’t do it. Rule 2 (for experts only). Don’t do it yet—that is, not until you have a perfectly clear and unoptimized solution. " :D